woocommerce/woocommerce-android
# Remove login site address when creating new account from NoWPCom erro…

Fixes Log in bug 

### Description

While working on https://github.com/woocommerce/woocommerce-android/issues/9967 and testing different flows I noticed we have a bug where we are not clearing the entered site address when `LoginNoWPcomAccountFoundDialogFragment` is displayed and user selects `Create An Account`. This will lead to user creating a new account and the app displaying the account mismatch error screen all the time. 

## Bug

https://github.com/woocommerce/woocommerce-android/assets/2663464/e3742186-2a1f-4b9e-b315-0580c8c5feca

This prevents the user from creating a store even after killing the app: 

https://github.com/woocommerce/woocommerce-android/assets/2663464/380d8731-b72d-4d45-ae86-5212270ae2f0


## Expected behavior

https://github.com/woocommerce/woocommerce-android/assets/2663464/01b7cfc5-f280-47ee-8a9a-cb7487fc108b

### Testing instructions

1. Log in using site address
2. Enter valid site WP.com site address (or Jetpack connected site address)
3. Enter a non WP.com email
4. Click on `Create An Account` from `LoginNoWPcomAccountFoundDialogFragment` 
5. Create a new account 
6. Check that store creation flow is displayed instead of `AccountMismatched` error
